.gitlab/scripts/publish_layers.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2024 Datadog, Inc.
+
+set -e
+
+LAYER_DIR=".layers"
+VALID_ACCOUNTS=("sandbox" "prod")
+
+publish_layer() {
+    region=$1
+    layer=$2
+    file=$3
+    compatible_architectures=$4
+
+    version_nbr=$(aws lambda publish-layer-version --layer-name $layer \
+        --description "dd-trace-dotnet" \
+        --compatible-runtimes "dotnet6" "dotnet8" \
+        --compatible-architectures $compatible_architectures \
+        --zip-file "fileb://${file}" \
+        --region $region \
+        | jq -r '.Version'
+    )
+
+    # Add permissions only for prod
+    if [ "$STAGE" == "prod" ]; then
+        permission=$(aws lambda add-layer-version-permission --layer-name $layer \
+            --version-number $version_nbr \
+            --statement-id "release-$version_nbr" \
+            --action lambda:GetLayerVersion \
+            --principal "*" \
+            --region $region
+        )
+    fi
+
+    echo $version_nbr
+}
+
+
+if [ -z "$ARCHITECTURE" ]; then
```

Comment:
same here, lots of duplicate code that can be reusable functions.
Arg validation could be like https://github.com/DataDog/datadog-lambda-extension/blob/main/bottlecap-run/bottlecap_dev.sh#L8-L15

Reply:
I agree, but definitely not doing it for this PR, this should be done using the `serverless-tools` repo eventually, as a CLI tool of our own